packages/core/src/render3/instructions/styling.ts

```diff
@@ -302,11 +306,13 @@ export function classMapInternal(
   // there should not be a situation where a directive host bindings function
   // evaluates the inputs (this should only happen in the template function)
   if (!isHostStyling() && hasDirectiveInput && classes !== NO_CHANGE) {
-    updateDirectiveInputValue(context, lView, tNode, bindingIndex, classes, true);
+    updateDirectiveInputValue(
+        context, lView, tNode, bindingIndex, classes, true, tView.firstUpdatePass);
```

Comment:
Same here. We can avoid reading the property twice

```suggestion
        context, lView, tNode, bindingIndex, classes, true, firstUpdatePass);
```

Reply:
fixed